monkey/agent_plugins/exploiters/ssh/src/ssh_client.py

```diff
@@ -27,39 +27,67 @@ def __init__(self):
     def connect(
         self, host: TargetHost, credentials: Credentials, port: NetworkPort, timeout: float
     ):
-        self._client = paramiko.SSHClient()
-        self._client.set_missing_host_key_policy(paramiko.WarningPolicy())
+        """
+        Connect to the host using SSH
+
+        Credentials may be a username and password, or a username and private key.
+
+        :param host: Host to connect to
+        :param credentials: Credentials to use for the connection
+        :param port: Port to connect to
+        :param timeout: Timeout for the connection, in seconds
+        :raises Exception: If the connection could not be established
+        """
+
+        username = (
+            credentials.identity.username if isinstance(credentials.identity, Username) else None
+        )
```

Comment:
Can paramiko connect if username is `None`? The docstring says, "Credentials may be a username and password, or a username and private key."

Reply:
Probably not :)